src/features/messages/cards/TransactionCard.tsx

```diff
@@ -236,11 +249,43 @@ function CallDataModal({ debugResult }: { debugResult?: MessageDebugResult }) {
           </p>
           <LabelAndCodeBlock label="Recipient contract address:" value={contract} />
           <LabelAndCodeBlock label="Handle function input calldata:" value={handleCalldata} />
+          <button onClick={handleClick}
+            disabled={disabled}
+            className='underline text-blue-400'
+          >
+            {buttonText}
+          </button>
+          {showSpinner && <Spinner classes="mt-4 scale-75 self-center" />}
         </div>
       </Modal>
     </>
   );
 }
+async function simulateCall({contract,handleCalldata,chainId,message}:{contract:string,handleCalldata:string,chainId:ChainId,message:Message}){
+
+  const data={
+        save: true,
+        save_if_fails: true, 
+        simulation_type: 'full',
+        network_id: chainId, 
```

Comment:
rename to chainId to be consistent with naming elsewhere

Reply:
the keys cant be changed inside the body object or else the post request fails
same goes for changing them to camelCase

Reply:
Sorry, I realize now that these must match Tenderly's schema. Please ignore.